test/e2e/metrics_test.go

```diff
@@ -41,14 +49,16 @@ func TestOperatorControllerMetricsExportedEndpoint(t *testing.T) {
 		"https://operator-controller-service.NAMESPACE.svc.cluster.local:8443/metrics",
 	)
 
-	config.run()
+	mtc.run()
 }
 
 // TestCatalogdMetricsExportedEndpoint verifies that the metrics endpoint for catalogd
 func TestCatalogdMetricsExportedEndpoint(t *testing.T) {
-	client := utils.FindK8sClient(t)
-	config := NewMetricsTestConfig(
-		t, client,
+	kubeClient, restConfig := findK8sClient(t)
```

Comment:
I was thinking that since metrics endpoint tests are part of the general e2e suite, we could also use the `client.Client` that we initialize as part of `TestMain` entrypoint:
```go
func TestMain(m *testing.M) {
	cfg = ctrl.GetConfigOrDie()

	var err error
	utilruntime.Must(apiextensionsv1.AddToScheme(scheme.Scheme))
	c, err = client.New(cfg, client.Options{Scheme: scheme.Scheme})
	utilruntime.Must(err)

	os.Exit(m.Run())
}
```

For that to work, we'd have to rename it, as it's currently available on a package-level as `c` and would be shadowed by `MetricsTestConfig` receiver name, which is also called `c` atm. We could also rename the receiver ofc ;)

WDYT?

Reply:
yeah, makes sense to me, my solution is to rename the main test one: `globalClient`